test/conformance/tests/sriov_operator.go

```diff
@@ -139,61 +140,28 @@ var _ = Describe("[sriov] operator", func() {
 		resourceName := "testresource"
 		var node string
 		var sriovDevice *sriovv1.InterfaceExt
+		var discoveryModeSkip bool
 
 		execute.BeforeAll(func() {
-			node = sriovInfos.Nodes[0]
 			var err error
-			sriovDevice, err = sriovInfos.FindOneSriovDevice(node)
-			Expect(err).ToNot(HaveOccurred())
 
-			// For the context of tests is better to use a Mellanox card
-			// as they support all the virtual function flags
-			// if we don't find a Mellanox card we fall back to any sriov
-			// capability interface and skip the rate limit test.
-			intf, err := sriovInfos.FindOneMellanoxSriovDevice(node)
-			if err != nil {
-				intf, err = sriovInfos.FindOneSriovDevice(node)
-				Expect(err).ToNot(HaveOccurred())
-			}
+			if discovery.Enabled() {
+				node, resourceName, numVfs, err = discovery.DiscoveredResources(clients, sriovInfos.Nodes, operatorNamespace)
+				discoveryModeSkip = node == "" || resourceName == "" || numVfs < 5 || err != nil
```

Comment:
Expect err to not be nil.
Also, makes sense to provide a reason for why we skipped

Reply:
Done